persistence/src/main/java/se/inera/intyg/intygstjanst/persistence/model/dao/impl/RelationDaoImpl.java

```diff
@@ -63,6 +63,21 @@ public List<Relation> getParent(String intygsId) {
             .getResultList();
     }
 
+    @Override
+    public List<Relation> getRelations(List<String> certificateIds, List<String> relationCodes) {
+        final var query = entityManager.createQuery(
+                "SELECT r FROM Relation r WHERE r.toIntygsId IN :certificateIds OR r.fromIntygsId IN :certificateIds"
+                        + "AND r.relationKod IN :relationCodes", Relation.class
+                )
+                .setParameter("certificateIds", certificateIds);
+
+        if (relationCodes != null && !relationCodes.isEmpty()) {
```

Comment:
Is this fine or will it throw an error if relationCodes is not set but used in query

Reply:
Fixed this problem.